clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7309,6 +7309,9 @@ let CategoryName = "Lambda Issue" in {
     "here">;
   def note_var_explicitly_captured_here : Note<"variable %0 is"
     "%select{| explicitly}1 captured here">;
+  def err_implicit_this_capture : Error<
+    "implicit capture of 'this' is not allowed for kernel functions in "
+    "SYCL 1.2.1">;
```

Comment:
Is there a reason you're saying SYCL 1.2.1. Are there plans to allow this later? I think the diagnostic can just say implicit capture of 'this' is not allowed in kernel function,  especially since the code does not seem to be checking std/spec version, etc.

Reply:
Thanks @elizabethandrews. Intent was to give the programmer a reference to look for further clarification. I do see a few other warning/error messages using a similar format. But I have no preference on this matter though. @erichkeane, can you also please provide your feedback? Thanks.

Reply:
I don't think there is value in mentioning the SYCL version. The only times we do this sort of thing is for the c++98-compat warning group (and similar warnings).

I'd say just cut the error message after 'functions'.